integration-cli/docker_cli_build_test.go

```diff
@@ -669,6 +669,64 @@ func TestBuildCacheADD(t *testing.T) {
 	logDone("build - build two images with remote ADD")
 }
 
+func TestBuildLastModified(t *testing.T) {
+	name := "testbuildlastmodified"
+	defer deleteImages(name)
+
+	server, err := fakeStorage(map[string]string{
+		"file": "hello",
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer server.Close()
+
+	var out, out2 string
+
+	if _, out, err = buildImageWithOut(name,
+		fmt.Sprintf("FROM busybox\nADD %s/file /\nRUN ls -le /file", server.URL),
+		false); err != nil {
+		t.Fatal(err)
+	}
+
+	originMTime := regexp.MustCompile(`root.*/file.*\n`).FindString(out)
+	// Make sure our regexp is correct
+	if strings.Index(originMTime, "/file") < 0 {
+		t.Fatalf("Missing ls info on 'file':\n%s", out)
+	}
+
+	// Build it again and make sure the mtime of the file didn't change.
+	// Wait a few seconds to make sure the time changed enough to notice
+	time.Sleep(2 * time.Second)
+
+	if _, out2, err = buildImageWithOut(name,
+		fmt.Sprintf("FROM busybox\nADD %s/file /\nRUN ls -le /file", server.URL),
+		false); err != nil {
+		t.Fatal(err)
+	}
+
+	newMTime := regexp.MustCompile(`root.*/file.*\n`).FindString(out2)
+	if newMTime != originMTime {
+		t.Fatalf("MTime changed:\nOrigin:%s\nNew:%s", originMTime, newMTime)
+	}
+
+	// Now 'touch' the file and make sure the timestamp DID change this time
+	server.FakeContext.Add("file", "bye")
```

Comment:
this is definitely not gonna work in the case of windows CI since the daemon is not at the same host and we're trying to serve those files over at the remote daemon host in #10893. so this change won't be reflected to the remote host which already has started a container by COPYing this context over there.